### DISCUSSION TITLE
Smart home integration

### DISCUSSION THREAD

**Discussion by adocampo** [OpenVoiceOS/ovos-buildroot#94] [Ideas]
Alexa echo devices has a cool feature I think OVOS could mimic.
When swipping left, an interesting menu appears, in it, we can see the items Communications, Digital Home, Music, Videos, Routines, and Alarms.
Yet all of them can be of use, which I find more useful perhaps is the Digital Home, which consist in a set of switches and groups Alexa can control.
I use Mycroft with Home Assitant, and I think there are other smart home integrations skills out there. 
It would be super cool to have a set of switches of all mycroft can control right there, for those times one cannot speak out loud or there are environmental noise which makes it difficult for mycroft to understand.

**Comment by ChanceNCounter** [comment]
Personally, I would rather do a big push toward the HiveMind protocol, and then instead of writing Mycroft or OVOS integrations for each of the smart home systems, we can write translation layers. Intent <---> Action, just like with Mycroft. That's the theory, anyway.

So, possibly out of scope for project, definitely in scope for devs? XD

**Comment by adocampo** [comment]
I had to buy a Xiaomi POCO X3 lately, and I was amazed by its "native" integration. MIUI detects all your domotic and smart home devices (thanks to the corresponding app installed on the phone), and shows under the fast switches all the possible interactions for those devices from, in my case, home assistant app
![1634884440259](https://user-images.githubusercontent.com/2266612/138404951-c4983273-b3e1-4b7a-8b77-8a81c946eb88.jpg)
![Screenshot_2021-10-22-08-32-52-378_com mi android globallauncher](https://user-images.githubusercontent.com/2266612/138404944-f52970c8-0c3c-4dc5-8b34-7b8a73d7de82.jpg)

I was thinking in something similar, I mean, prepare OVOS GUI to allocate all those switches taken from home-assistant (et al) skill or the translation layer you're speaking of, and let the user control the devices right from there (turning on/off switches, controlling LED intensity, running scenes or scripts, etc)

**Reply by ChanceNCounter** [reply]
I agree, it just probably isn't a system job.

**Comment by AIIX** [comment]
The UI layer isn't the issue there needs to be a backend that supports this currently afaik there isn't any in Mycroft itself, If one was to look into developing this integration I would suggest looking into the Matter API or the HomeAssistant  API via which this could be made possible

**Reply by ChanceNCounter** [reply]
Jarbas raised the notion in chat of making HomeAssistant a priority for HiveMind integration, which would sorta bring it full circle =P

**Reply by adocampo** [reply]
What to say... voice assistants are essential for home automation, in my opinion. Sure, they can do a lot more than turning things on and off, but controlling all the IoT devices from a phone or tablet is... well, not more useful than having an old and error-free switch.

Hivemind did work on a rpi zero w, didn't it? I think it's a good time to buy one or two of them  😛

**Reply by ChanceNCounter** [reply]
yep, the voice sat worked like a charm. we're now discussing the chicken in dev chat, so as to attract hens. i might get my stupid doorbell!

**Comment by j1nx** [comment]
The boxes (easy accessible  cards) are being implemented;
https://user-images.githubusercontent.com/19663666/157872173-e77b70ca-7d2e-4039-a51d-2393835fb657.gif

However till we have migrated the "Common IOT framework" over to ovos-core and possible associated IOT skills (homeassistant) there are no boxes to be shown.

**Reply by NeonDaniel** [reply]
RE: Those cards, I had some ideas about the implementation on the code side (inspired by Matrix discussion).
* A decorator on an intent handler could specify how it is called via a GUI element (GUI would just send a message with some specified contents)
* This could be a separate skill for "voice shortcuts", maybe [skill-synonyms](https://github.com/NeonGeckoCom/skill-synonyms) or something similar. The skill basically captures some user utterance and sends out a different utterance to the Intent Service based on configuration. Might even be able to define basic cards via intent?
* I'm not sure how this decorator would be used to update card data, maybe an updater function arg?
